core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -160,6 +160,50 @@ class TransactionsTest extends KafkaServerTestHarness {
   }
 
   @Test
+  def testDelayedFetchIncludesAbortedTransaction(): Unit = {
```

Comment:
The test case covers this. I'm planning to spend a bit more time with the `ReplicaManager` tests after this patch gets in. For something so critical, the coverage seems insufficient.